zypp/target/rpm/RpmDb.cc

```diff
@@ -1160,60 +1172,80 @@ namespace
     rpmts ts = ::rpmtsCreate();
     ::rpmtsSetRootDir( ts, root_r.c_str() );
     ::rpmtsSetVSFlags( ts, RPMVSF_DEFAULT );
-
-    rpmQVKArguments_s qva;
-    memset( &qva, 0, sizeof(rpmQVKArguments_s) );
-#ifndef HAVE_RPM_VERIFY_TRANSACTION_STEP
-    // Legacy: In rpm >= 4.15 qva_flags symbols don't exist
-    // and qva_flags is not used in signature checking at all.
-    qva.qva_flags = (VERIFY_DIGEST|VERIFY_SIGNATURE);
-#else
+#ifdef HAVE_RPM_VERIFY_TRANSACTION_STEP
     ::rpmtsSetVfyFlags( ts, RPMVSF_DEFAULT );
 #endif
+
     RpmlogCapture vresult;
     LocaleGuard guard( LC_ALL, "C" );	// bsc#1076415: rpm log output is localized, but we need to parse it :(
+    static rpmQVKArguments_s qva = ([](){ rpmQVKArguments_s qva; memset( &qva, 0, sizeof(rpmQVKArguments_s) ); return qva; })();
```

Comment:
Just a comment: We need to be careful with the use of non const statics, in cases where it makes sense to use threadpool for certain jobs this can be a problem.  Also many statics increase our memory footprint.

Reply:
( THis is not a threadpool candidate because we call into librpm here, just wanted to point it out for future PRs )